src/cdk-lib/amazonaurora/aurora-default-vector-store.ts

```diff
@@ -0,0 +1,257 @@
+  * private with NAT Gateway, private without NAT Gateway),
+  * with the Amazon Aurora Serverless V2 Cluster.
+  * The cluster has 1 writer/reader of PostgreSQL version 15.5
+  * instance (min capacity 0.5, max capacity 4). Lambda custom
+  * resource executes required pgvector and Amazon Bedrock Knowledge
+  * Base SQL queries against Aurora cluster
+  * during deployment. The secret containing databases credentials is
+  * being deployed and securely stored in AWS Secrets Manager.
+  * You must specify the same embeddings model that you used in
+  * KnowledgeBase construct.
+  * @see https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.VectorDB.html)
+  */
+export class AmazonAuroraDefaultVectorStore extends cdk.Resource {
+  /**
+   * The ARN of your Amazon Aurora DB cluster.
+   */
+  public readonly resourceArn: string;
+
+  /**
+   * The name of your Database.
+   */
+  public readonly databaseName: string;
+
+  /**
+   * The Table Name of your Amazon Aurora DB cluster.
+   */
+  public readonly tableName: string;
+
+  /**
+   * The Secret ARN of your Amazon Aurora DB cluster.
+   */
+  public readonly credentialsSecretArn: string;
+
+  /**
+   * Primary key of your Amazon Aurora DB cluster.
+   */
+  public readonly primaryKeyField: string;
+
+  /**
+   * Cluster identifier of your Amazon Aurora DB cluster.
+   */
+  public readonly clusterIdentifier: string;
+
+  /**
+   * Model used for embeddings.
+   */
+  public readonly embeddingsModel: BedrockFoundationModel;
+
+  /**
+   * An IAM policy that allows Data API access to Aurora.
+   * @private
+   */
+  private auroraPgCRPolicy: iam.ManagedPolicy;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    props: AmazonAuroraDefaultVectorStoreProps,
+  ) {
+    super(scope, id);
+
+    this.databaseName = 'bedrock_vector_db';
+    this.tableName = 'bedrock_integration.bedrock_kb';
+    this.primaryKeyField = 'id';
+    this.clusterIdentifier = 'aurora-serverless-vector-cluster';
+    this.embeddingsModel = props.embeddingsModel;
+
+    const vpc = buildVpc(this, {});
+    vpc.addFlowLog('VpcFlowLog', {
+      destination: ec2.FlowLogDestination.toCloudWatchLogs(),
+    });
+
+    const auroraSecurityGroup = new ec2.SecurityGroup(this, 'AuroraSecurityGroup', {
+      vpc: vpc,
+      securityGroupName: 'aurora-security-group',
+      description: 'Security group for access to Aurora from Lambda',
+    });
+
+    const lambdaSecurityGroup = new ec2.SecurityGroup(this, 'LambdaSecurityGroup', {
+      vpc: vpc,
+      securityGroupName: 'lambda-security-group',
+      description: 'Security group for Lambda access to Aurora',
+    });
+
+    const auroraCluster = new rds.DatabaseCluster(this, 'AuroraCluster', {
+      engine: rds.DatabaseClusterEngine.auroraPostgres({
+        version: rds.AuroraPostgresEngineVersion.VER_15_5,
+      }),
+      credentials: rds.Credentials.fromGeneratedSecret('postgres'),
```

Comment:
when do we create this key ' AmazonAuroraDefaultVectorStoreProps' in secrets

Reply:
Not sure I am following the question ... Do you mean where do we get credentials from? 
`fromGeneratedSecret` will create a secret with username `postgres` for us. Please see [here](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.Credentials.html#static-fromwbrgeneratedwbrsecretusername-options) for details.